File: frontend/src/api/reports.ts
// reports API client (09_API_SPEC §8). PRD Patch #5: soft delete + restore.

import {
  apiDelete,
  apiGet,
  apiGetList,
  apiPatch,
  apiPost,
  type ApiListResponse,
} from '@/lib/api';

export type ReportStatusLit =
  | 'PENDING'
  | 'GENERATING'
  | 'COMPLETED'
  | 'FAILED';

export interface ReportOut {
  id: string;
  stock_code: string;
  stock_name: string;
  status: ReportStatusLit;
  model_version: string | null;
  prompt_tokens: number | null;
  completion_tokens: number | null;
  narrative_part: string | null;
  facts_part: string | null;
  data_sources: Record<string, unknown> | null;
  pdf_path: string | null;
  excel_path: string | null;
  user_notes: string | null;
  error_message: string | null;
  progress: number | null;
  elapsed_seconds: number | null;
  generation_started_at: string | null;
  generation_completed_at: string | null;
  generation_duration_seconds: number | null;
  requested_at: string;
  created_at: string;

  // ★ PRD Patch #5: soft-delete
  is_hidden: boolean;
  hidden_at: string | null;
  hidden_reason: string | null;
}

export interface ReportRequestBody {
  stock_code: string;
  tracked_stock_id?: string | null;
}

export interface ReportRequestResponse {
  report_id: string;
  status: ReportStatusLit;
  estimated_seconds: number;
  stock_code: string;
  stock_name: string;
  requested_at: string;
}

export interface ReportPatch {
  user_notes?: string | null;
}

export interface ReportListParams {
  stock_code?: string;
  status?: ReportStatusLit;
  from_date?: string;
  to_date?: string;
  include_hidden?: boolean;  // ★ PRD Patch #5
  limit?: number;
  cursor?: string;
}

export const reportsApi = {
  list(params: ReportListParams = {}): Promise<ApiListResponse<ReportOut>> {
    return apiGetList<ReportOut>('/api/v71/reports', { params });
  },
  get(id: string): Promise<ReportOut> {
    return apiGet<ReportOut>(`/api/v71/reports/${id}`);
  },
  request(body: ReportRequestBody): Promise<ReportRequestResponse> {
    return apiPost<ReportRequestResponse, ReportRequestBody>(
      '/api/v71/reports/request',
      body,
    );
  },
  patch(id: string, body: ReportPatch): Promise<ReportOut> {
    return apiPatch<ReportOut, ReportPatch>(`/api/v71/reports/${id}`, body);
  },
  // ★ PRD Patch #5: soft delete (영구 보존, 목록에서만 숨김)
  remove(id: string): Promise<void> {
    return apiDelete(`/api/v71/reports/${id}`);
  },
  // ★ PRD Patch #5: 숨긴 리포트 복구
  restore(id: string): Promise<ReportOut> {
    return apiPost<ReportOut>(`/api/v71/reports/${id}/restore`);
  },
};

File: src/web/v71/schemas/reports.py
"""Report schemas (09_API_SPEC §8)."""

from __future__ import annotations

from datetime import datetime
from typing import Any, Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

ReportStatusLit = Literal["PENDING", "GENERATING", "COMPLETED", "FAILED"]


class ReportRequest(BaseModel):
    """09_API_SPEC §8.1."""

    stock_code: str = Field(min_length=6, max_length=10, pattern=r"^\d{6,10}$")
    tracked_stock_id: UUID | None = None


class ReportRequestResponse(BaseModel):
    report_id: UUID
    status: ReportStatusLit
    estimated_seconds: int = 300
    stock_code: str
    stock_name: str
    requested_at: datetime


class ReportOut(BaseModel):
    """09_API_SPEC §8.2 + §8.3."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    stock_code: str
    stock_name: str
    status: ReportStatusLit
    model_version: str | None = None
    prompt_tokens: int | None = None
    completion_tokens: int | None = None
    narrative_part: str | None = None
    facts_part: str | None = None
    data_sources: dict[str, Any] | None = None
    pdf_path: str | None = None
    excel_path: str | None = None
    user_notes: str | None = None
    error_message: str | None = None
    progress: int | None = None  # populated when GENERATING
    elapsed_seconds: int | None = None
    generation_started_at: datetime | None = None
    generation_completed_at: datetime | None = None
    generation_duration_seconds: int | None = None
    requested_at: datetime
    created_at: datetime

    # ★ PRD Patch #5 (V7.1.0d, 2026-04-27): soft-delete metadata.
    is_hidden: bool = False
    hidden_at: datetime | None = None
    hidden_reason: str | None = None


class ReportPatch(BaseModel):
    """09_API_SPEC §8.6 -- user_notes only."""

    user_notes: str | None = Field(default=None, max_length=10000)


class ReportListParams(BaseModel):
    """09_API_SPEC §8.3 query params (PRD Patch #5: include_hidden 추가)."""

    stock_code: str | None = None
    status: ReportStatusLit | None = None
    from_date: str | None = None
    to_date: str | None = None
    include_hidden: bool = False  # ★ PRD Patch #5
    limit: int = Field(default=20, ge=1, le=100)
    cursor: str | None = None
